### DISCUSSION TITLE
`cub::DeviceRadixSort` uses Policy900 when compiled for an A100 GPU

### DISCUSSION THREAD

**Discussion by mfbalin** [NVIDIA/cccl#1241] [CUB] [answered]
Hi everyone,

In https://github.com/dmlc/dgl/pull/6770, we utilize `cub::DeviceRadixSort` and we profiled the execution after compiling for sm_80 to run on A100 GPU. However, it can be seen that the `Policy900` was used in the profile below. Is this normal? I would have expected Policy800 to be selected for the A100 GPU.

![image](https://github.com/NVIDIA/cccl/assets/16190101/2dc27056-0cee-49b0-92f6-cf1f4f68e895)

**Comment by gevtushenko** [comment, marked as answer]
Hello @mfbalin! 

If you are referring to the symbol name having Policy900 in it, it's due to the fact that we pass **maximal** policy (`DeviceRadixSortOnesweepKernel<MaxPolicyT...`) to the kernel and then statically dispatch to *active* policy inside the kernel. The description of chained policy can be found in our [docs](https://nvidia.github.io/cccl/cub/developer_overview.html#dispatch-layer).

**Reply by mfbalin** [reply]
I see, thank you for the quick reply, that is good to know.
